app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -974,6 +974,14 @@ private void storeSharedPrefs(Place selectedPlace) {
         applicationKvStore.putJson(PLACE_OBJECT, selectedPlace);
     }
 
+    /**
+     * This function clears the place data from the application, so that when the user leaves the app and
+     * tries to upload an other image the data is not kept (see #2803)
+     */
+    public void removeSharedPrefs() {
```

Comment:
This also can be implemented in NearbyParentPresenter, instead of fragment. Since this is not a UI related task and according to MVP it should not be implemented here.